drivers/tee/optee/optee.c

```diff
@@ -268,6 +268,40 @@ static uint32_t optee_call_supp(const struct device *dev, uint32_t func, size_t
 
 	return ret;
 }
+
+static int cmd_alloc_suppl(const struct device *dev, size_t sz, struct tee_shm **shm)
+{
+	uint32_t ret;
+	struct tee_param param;
+
+	param.attr = TEE_PARAM_ATTR_TYPE_VALUE_INOUT;
+	param.a = OPTEE_RPC_SHM_TYPE_APPL;
+	param.b = sz;
+	param.c = 0;
+
+	ret = optee_call_supp(dev, OPTEE_RPC_CMD_SHM_ALLOC, 1, &param);
+
+	if (ret)
+		return ret;
+
+	ret = tee_add_shm(dev, (void *)param.c, 0, param.b, 0, shm);
+
+	return ret;
+}
+
+static void cmd_free_suppl(const struct device *dev, struct tee_shm *shm)
+{
+	struct tee_param param;
+
+
+	param.attr = TEE_PARAM_ATTR_TYPE_VALUE_INOUT;
```

Comment:
may be done in initialization:
`... param = {
    .attr = ...
};`

Reply:
note: here can be a leak of some data, also if we have more fields that you initialize here some of fields will contain garbage and it can influence working of OPTEE OS

initializers are like: `{}`, `{0}` and `{.attr = ...}` solve the last issue, but they still have security issues in case when we have paddings between fields in structure, they can contain some stack garbage and it can be useful for some hacker's attacks

so, the best way is to use here `memset` for `param`
but I don't see how somebody can use stack garbage in structure paddings in this case